configure.ac

```diff
@@ -5387,7 +5387,33 @@ PKG_CHECK_MODULES([LIBLZMA], [liblzma], [have_liblzma=yes], [
 ])
 
 dnl zstd 1.4.5 stabilised ZDICT_finalizeDictionary
-PKG_CHECK_MODULES([LIBZSTD], [libzstd >= 1.4.5], [have_libzstd=yes], [have_libzstd=no])
+PKG_CHECK_MODULES([LIBZSTD], [libzstd >= 1.4.5], [have_libzstd=yes], [
+  WITH_SAVE_ENV([
+    CPPFLAGS="$CPPFLAGS $LIBZSTD_CFLAGS"
+    CFLAGS="$CFLAGS $LIBZSTD_CFLAGS"
+    LIBS="$LIBS $LIBZSTD_LIBS"
+    AC_SEARCH_LIBS([ZDICT_finalizeDictionary], [zstd], [
+      AC_MSG_CHECKING([ZSTD_VERSION_NUMBER >= 1.4.5])
+      AC_COMPILE_IFELSE([
```

Comment:
I think we should check for zstd.h before doing a compilation check. If we don't have the headers available then we can skip the check entirely

Reply:
I updated https://github.com/python/cpython/pull/133550 to include the COMPILE_IFELSE and MSG_CHECKING changes, but using the order of checks in that PR.

Reply:
The reason I swapped them is that `AC_CHECK_HEADERS` performs action-if-true for every header in the list, leading to duplicated work. So I think it's somewhat faster to invert the checks.


cc @erlend-aasland for advice